msal/src/test/java/com/microsoft/identity/client/e2e/tests/network/nativeauth/NativeAuthPublicClientApplicationAbstractTest.kt

```diff
@@ -131,4 +131,28 @@ abstract class NativeAuthPublicClientApplicationAbstractTest : IPublicClientAppl
             Assert.fail(e.message)
         }
     }
+
+    fun <T> retryOperation(
+        maxRetries: Int = 3,
+        onFailure: () -> Unit = { Assert.fail() },
+        authFlow: () -> T
+    ) {
+        var retryCount = 0
+        var shouldRetry = true
+
+        while (shouldRetry) {
+            try {
+                authFlow()
+                shouldRetry = false // authFlow() has succeeded, so we don't need to retry.
+            } catch (e: IllegalStateException) {
```

Comment:
Is this the only possible exception type that could be thrown? Should we catch a broader range for unexpected exceptions?

Reply:
We should only catch expected exceptions (in this case, related to the retry). Catching all/too many exceptions will cause unexpected exceptions (due to bugs or other side effects we're not aware of) to also be caught and not make the test fail, which is not what we want.